### PR TITLE
Debunk some myths

### PR DESCRIPTION
**Pull request by mikearnaldi**

### CODE REVIEW

File: content/docs/700-other/150-myths.mdx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+Effect's internals are not built on generators, we only use generators to provide an API which closely mimics async-await. Internally async-await uses the same mechanics as generators and they are equally performant. So if you don't have a problem with async-await you won't have a problem with Effect's generators.
+
+Where generators and iterables are unacceptably slow is in transforming collections of data, for that try to use plain arrays as much as possible.
+
+## Effect will make your code 500x slower!
+
+That's pure FUD!
+
+Effect does perform 500x slower if you are comparing:
```
Comment (fubhy) [inline content/docs/700-other/150-myths.mdx:17]:
```suggestion
Effect does perform 500x slower if you are comparing:
```

File: content/docs/700-other/150-myths.mdx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+- Effect.sync
+- Effect.tryPromise
+- Effect.gen
+- Effect.runPromise
+- Effect.catchTag
+- Effect.catchAll
+- Effect.acquireRelease
+- Effect.acquireUseRelease
+- Effect.provide
+- Effect.provideService
```
Comment (fubhy) [inline content/docs/700-other/150-myths.mdx:76]:
```suggestion
- Effect.succeed
- Effect.fail
- Effect.sync
- Effect.tryPromise
- Effect.gen
- Effect.runPromise
- Effect.catchTag
- Effect.catchAll
- Effect.acquireRelease
- Effect.acquireUseRelease
- Effect.provide
- Effect.provideService
- Effect.andThen
- Effect.map
- Effect.tap
```

**Review by fubhy — CHANGES_REQUESTED** [review]

File: content/docs/700-other/150-myths.mdx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+A short list of commonly used modules:
+
+- Effect
+- Context
+- Layer
+- Option
+- Either
+- Array
+- Match
```
Comment (fubhy) [inline content/docs/700-other/150-myths.mdx:86]: I'd probably leave out `Match` here, wdyt?

**Review by fubhy — CHANGES_REQUESTED** [review]

File: content/docs/700-other/150-myths.mdx (hunk 3 above)
Comment (mikearnaldi) [reply content/docs/700-other/150-myths.mdx:86]: Given ts-pattern I think that would fall into mainstream interest

Comment (fubhy) [reply content/docs/700-other/150-myths.mdx:86]: Fair. Personally, I don't use it that much on a regular basis compared to the others listed. But that's subjective I guess.

File: content/docs/700-other/150-myths.mdx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+## Effect is the same as RxJS and shares its problems
+
+This is a sensitive topic, let's start by saying that RxJS is a great project and that it has helped millions of developers write reliable software and we all should be thankful to the developers who contributed to such an amazing project.
+
+Discussing the scope of the projects, RxJS aims to make working with Observables easy and wants to provide reactive extensions to JS, Effect instead wants to make writing production-grade TypeScript easy. While the intersection is non-empty the projects have fundamentally different objectives and strategies.
+
+Sometimes people refer to RxJS in bad light, and the reason isn't RxJS in itself but rather usage of RxJS in problem domains where RxJS wasn't thought to be used.
+
+Namely the idea that "everything is a stream" is theoretically true but it leads to foundamental limitations on developer experience, the primary issue being that streams are multi-shot (emit potentially multiple elements, or zero) and mutable delimited continuations (JS Generators) are known to be only good to represent single-shot effects (that emit a single value).
```
Comment (fubhy) [inline content/docs/700-other/150-myths.mdx:97]:
```suggestion
Namely the idea that "everything is a stream" is theoretically true but it leads to fundamental limitations on developer experience, the primary issue being that streams are multi-shot (emit potentially multiple elements, or zero) and mutable delimited continuations (JS Generators) are known to be only good to represent single-shot effects (that emit a single value).
```

**Review by fubhy — CHANGES_REQUESTED** [review]

**Pull request merged**